yazi-core/src/folder/files.rs

```diff
@@ -46,7 +46,11 @@ impl Deref for Files {
 
 impl Files {
 	pub async fn from_dir(url: &Url) -> std::io::Result<UnboundedReceiver<File>> {
-		let mut it = fs::read_dir(url).await?;
+		let mut it = fs::read_dir(url).await.map_err(|e| {
```

Comment:
It's unnecessary because `Files::assert_stale` has already checked it:

https://github.com/sxyazi/yazi/blob/d6081fbe6f57c8fa4b577fb3fbe1d392c057d94e/yazi-core/src/tab/preview.rs#L57-L58

Reply:
This is needed at least for "Permission denied" on a directory. `assert_stale` currently only checks for the metadata, not accessing the contents.